spec/formatting.md

```diff
@@ -216,7 +217,7 @@ the following steps are taken:
 
 4. Call the function implementation with the following arguments:
 
-   - The current _locale_.
+   - The current _formatting context_.
```

Comment:
This is too much, and potentially allows for too much "magic" in a custom function. I agree that the current text is missing the following:
- base direction
- bidi strategy
- locale matcher strategy (not referenced in our current spec, but relevant at least for JS)

The full function context also includes e.g. all of the raw input values, and those should definitely not be made all ä available to all functions.

Reply:
The formatting context might contain implemetation-specific goo (at Amazon we squirreled which site the user was on into field(s) in the context). 

I agree that a badly behaved function might do Bad Magick. But I think it's useful to give developers access to what they need and stay out of their way after that. You'll note that I made the exact same suggestion recently (but changed it at your request, because it was not a parsimonious change at that moment in time).

Reply:
And Stas and others want to be able to have access, like :u:function gender-agreement=$var.

_locale_ is certainly wrong.

Reply:
Ah, I see what you're saying.

It's not the same thing. The variable in an option is, itself, subject to having its value resolved by the MF processor _before_ the function is itself invoked (we say so in the options section). So it doesn't apply to needing the formatting context. The function gets the resolved value, not a variable reference.